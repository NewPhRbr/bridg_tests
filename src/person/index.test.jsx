import { render } from "@testing-library/react";
import { HttpResponse, http } from "msw";
import { setupServer } from "msw/node";
import { MemoryRouter } from "react-router-dom";
import { afterAll, afterEach, beforeAll, expect, test } from "vitest";
import PersonIndexPage from "./index";

const DATA = [
  {
    sex: "M",
    birth_date: "2024-05-21",
    death_date: "2024-05-21",
    death_date_estimated_indicator: true,
    death_indicator: true,
    id: 1,
    primary_name: {
      use: "string",
      family: "John",
      given: "Doe",
      middle: "",
      patronymic: "",
      prefix: "",
      suffix: "",
      id: 2,
      full: "Doe John",
    },
  },
];

const server = setupServer(http.get("/api/persons", () => HttpResponse.json(DATA)));

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

test("page renders correctly", async () => {
  const tree = render(
    <MemoryRouter>
      <PersonIndexPage />
    </MemoryRouter>
  );
  await new Promise((r) => setTimeout(r, 0));
  expect(tree).toMatchSnapshot();
});
