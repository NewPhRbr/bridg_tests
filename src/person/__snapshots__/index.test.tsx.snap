// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`page renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          class="mb-1 fs-7"
          style="--bs-breadcrumb-divider: '>';"
        >
          <ol
            class="breadcrumb my-auto"
          >
            <li
              class="breadcrumb-item active undefined"
            >
              Persons
            </li>
          </ol>
        </nav>
        <div
          class="d-flex undefined"
        >
          <div
            class="flex-grow-1"
          >
            <div
              class="lead fs-4"
            >
              List of Persons
            </div>
          </div>
          <div
            class="ps-4"
          >
            <a
              class="btn btn-sm btn-outline-primary"
              href="/new"
            >
              New
            </a>
          </div>
        </div>
        <div
          class="my-3"
        >
          <div
            class="dt-container dt-bootstrap5 dt-empty-footer"
            id="DataTables_Table_0_wrapper"
          >
            <div
              class="row mt-2 justify-content-between"
            >
              <div
                class="col-md-auto me-auto "
              />
              <div
                class="col-md-auto ms-auto "
              />
            </div>
            <div
              class="row mt-2 justify-content-md-center"
            >
              <div
                class="col-12 "
              >
                <table
                  aria-describedby="DataTables_Table_0_info"
                  class="table dataTable"
                  id="DataTables_Table_0"
                  style="width: 0px;"
                >
                  <colgroup>
                    <col
                      data-dt-column="0"
                      style="width: 0px;"
                    />
                    <col
                      data-dt-column="1"
                      style="width: 0px;"
                    />
                    <col
                      data-dt-column="2"
                      style="width: 0px;"
                    />
                    <col
                      data-dt-column="3"
                      style="width: 0px;"
                    />
                    <col
                      data-dt-column="4"
                      style="width: 0px;"
                    />
                    <col
                      data-dt-column="8"
                      style="width: 0px;"
                    />
                    <col
                      data-dt-column="9"
                      style="width: 0px;"
                    />
                    <col
                      data-dt-column="12"
                      style="width: 0px;"
                    />
                    <col
                      data-dt-column="13"
                      style="width: 0px;"
                    />
                  </colgroup>
                  <thead>
                    <tr
                      role="row"
                    >
                      <th
                        aria-label="ID: Activate to invert sorting"
                        aria-sort="ascending"
                        class="dt-orderable-asc dt-orderable-desc dt-type-numeric dt-ordering-asc"
                        colspan="1"
                        data-dt-column="0"
                        rowspan="1"
                        tabindex="0"
                      >
                        <span
                          class="dt-column-title"
                          role="button"
                        >
                          ID
                        </span>
                        <span
                          class="dt-column-order"
                        />
                      </th>
                      <th
                        aria-label="Name usage: Activate to sort"
                        class="dt-orderable-asc dt-orderable-desc"
                        colspan="1"
                        data-dt-column="1"
                        rowspan="1"
                        tabindex="0"
                      >
                        <span
                          class="dt-column-title"
                          role="button"
                        >
                          Name usage
                        </span>
                        <span
                          class="dt-column-order"
                        />
                      </th>
                      <th
                        aria-label="Family name: Activate to sort"
                        class="dt-orderable-asc dt-orderable-desc"
                        colspan="1"
                        data-dt-column="2"
                        rowspan="1"
                        tabindex="0"
                      >
                        <span
                          class="dt-column-title"
                          role="button"
                        >
                          Family name
                        </span>
                        <span
                          class="dt-column-order"
                        />
                      </th>
                      <th
                        aria-label="Given name: Activate to sort"
                        class="dt-orderable-asc dt-orderable-desc"
                        colspan="1"
                        data-dt-column="3"
                        rowspan="1"
                        tabindex="0"
                      >
                        <span
                          class="dt-column-title"
                          role="button"
                        >
                          Given name
                        </span>
                        <span
                          class="dt-column-order"
                        />
                      </th>
                      <th
                        aria-label="Middle name: Activate to sort"
                        class="dt-orderable-asc dt-orderable-desc dt-type-numeric"
                        colspan="1"
                        data-dt-column="4"
                        rowspan="1"
                        tabindex="0"
                      >
                        <span
                          class="dt-column-title"
                          role="button"
                        >
                          Middle name
                        </span>
                        <span
                          class="dt-column-order"
                        />
                      </th>
                      <th
                        aria-label="Sex: Activate to sort"
                        class="dt-orderable-asc dt-orderable-desc"
                        colspan="1"
                        data-dt-column="8"
                        rowspan="1"
                        tabindex="0"
                      >
                        <span
                          class="dt-column-title"
                          role="button"
                        >
                          Sex
                        </span>
                        <span
                          class="dt-column-order"
                        />
                      </th>
                      <th
                        aria-label="Birth date: Activate to sort"
                        class="dt-orderable-asc dt-orderable-desc dt-right"
                        colspan="1"
                        data-dt-column="9"
                        rowspan="1"
                        tabindex="0"
                      >
                        <span
                          class="dt-column-title"
                          role="button"
                        >
                          Birth date
                        </span>
                        <span
                          class="dt-column-order"
                        />
                      </th>
                      <th
                        aria-label="Dead: Activate to sort"
                        class="dt-orderable-asc dt-orderable-desc dt-type-numeric"
                        colspan="1"
                        data-dt-column="12"
                        rowspan="1"
                        tabindex="0"
                      >
                        <span
                          class="dt-column-title"
                          role="button"
                        >
                          Dead
                        </span>
                        <span
                          class="dt-column-order"
                        />
                      </th>
                      <th
                        aria-label=""
                        class="dt-orderable-none dt-type-numeric"
                        colspan="1"
                        data-dt-column="13"
                        rowspan="1"
                      >
                        <span
                          class="dt-column-title"
                        />
                        <span
                          class="dt-column-order"
                        />
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td
                        class="dt-type-numeric sorting_1"
                      >
                        1
                      </td>
                      <td>
                        string
                      </td>
                      <td>
                        John
                      </td>
                      <td>
                        Doe
                      </td>
                      <td
                        class="dt-type-numeric"
                      />
                      <td>
                        M
                      </td>
                      <td
                        class="dt-right"
                      >
                        21.05.2024
                      </td>
                      <td
                        class="dt-type-numeric"
                      >
                        true
                      </td>
                      <td
                        class="dt-type-numeric"
                      >
                        <div>
                          <a
                            href="/persons/1"
                          >
                            <i
                              class="fa-solid fa-right-to-bracket"
                            />
                          </a>
                        </div>
                      </td>
                    </tr>
                  </tbody>
                  <tfoot />
                </table>
              </div>
            </div>
            <div
              class="row mt-2 justify-content-between"
            >
              <div
                class="col-md-auto me-auto "
              >
                <div
                  aria-live="polite"
                  class="dt-info"
                  id="DataTables_Table_0_info"
                  role="status"
                >
                  Showing 1 to 1 of 1 entry
                </div>
              </div>
              <div
                class="col-md-auto ms-auto "
              >
                <div
                  class="dt-paging paging_full_numbers"
                >
                  <ul
                    class="pagination"
                  >
                    <li
                      class="dt-paging-button page-item disabled"
                    >
                      <a
                        aria-controls="DataTables_Table_0"
                        aria-disabled="true"
                        aria-label="First"
                        class="page-link first"
                        data-dt-idx="first"
                        tabindex="-1"
                      >
                        «
                      </a>
                    </li>
                    <li
                      class="dt-paging-button page-item disabled"
                    >
                      <a
                        aria-controls="DataTables_Table_0"
                        aria-disabled="true"
                        aria-label="Previous"
                        class="page-link previous"
                        data-dt-idx="previous"
                        tabindex="-1"
                      >
                        ‹
                      </a>
                    </li>
                    <li
                      class="dt-paging-button page-item active"
                    >
                      <a
                        aria-controls="DataTables_Table_0"
                        aria-current="page"
                        class="page-link"
                        data-dt-idx="0"
                        href="#"
                        tabindex="0"
                      >
                        1
                      </a>
                    </li>
                    <li
                      class="dt-paging-button page-item disabled"
                    >
                      <a
                        aria-controls="DataTables_Table_0"
                        aria-disabled="true"
                        aria-label="Next"
                        class="page-link next"
                        data-dt-idx="next"
                        tabindex="-1"
                      >
                        ›
                      </a>
                    </li>
                    <li
                      class="dt-paging-button page-item disabled"
                    >
                      <a
                        aria-controls="DataTables_Table_0"
                        aria-disabled="true"
                        aria-label="Last"
                        class="page-link last"
                        data-dt-idx="last"
                        tabindex="-1"
                      >
                        »
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        class="mb-1 fs-7"
        style="--bs-breadcrumb-divider: '>';"
      >
        <ol
          class="breadcrumb my-auto"
        >
          <li
            class="breadcrumb-item active undefined"
          >
            Persons
          </li>
        </ol>
      </nav>
      <div
        class="d-flex undefined"
      >
        <div
          class="flex-grow-1"
        >
          <div
            class="lead fs-4"
          >
            List of Persons
          </div>
        </div>
        <div
          class="ps-4"
        >
          <a
            class="btn btn-sm btn-outline-primary"
            href="/new"
          >
            New
          </a>
        </div>
      </div>
      <div
        class="my-3"
      >
        <div
          class="dt-container dt-bootstrap5 dt-empty-footer"
          id="DataTables_Table_0_wrapper"
        >
          <div
            class="row mt-2 justify-content-between"
          >
            <div
              class="col-md-auto me-auto "
            />
            <div
              class="col-md-auto ms-auto "
            />
          </div>
          <div
            class="row mt-2 justify-content-md-center"
          >
            <div
              class="col-12 "
            >
              <table
                aria-describedby="DataTables_Table_0_info"
                class="table dataTable"
                id="DataTables_Table_0"
                style="width: 0px;"
              >
                <colgroup>
                  <col
                    data-dt-column="0"
                    style="width: 0px;"
                  />
                  <col
                    data-dt-column="1"
                    style="width: 0px;"
                  />
                  <col
                    data-dt-column="2"
                    style="width: 0px;"
                  />
                  <col
                    data-dt-column="3"
                    style="width: 0px;"
                  />
                  <col
                    data-dt-column="4"
                    style="width: 0px;"
                  />
                  <col
                    data-dt-column="8"
                    style="width: 0px;"
                  />
                  <col
                    data-dt-column="9"
                    style="width: 0px;"
                  />
                  <col
                    data-dt-column="12"
                    style="width: 0px;"
                  />
                  <col
                    data-dt-column="13"
                    style="width: 0px;"
                  />
                </colgroup>
                <thead>
                  <tr
                    role="row"
                  >
                    <th
                      aria-label="ID: Activate to invert sorting"
                      aria-sort="ascending"
                      class="dt-orderable-asc dt-orderable-desc dt-type-numeric dt-ordering-asc"
                      colspan="1"
                      data-dt-column="0"
                      rowspan="1"
                      tabindex="0"
                    >
                      <span
                        class="dt-column-title"
                        role="button"
                      >
                        ID
                      </span>
                      <span
                        class="dt-column-order"
                      />
                    </th>
                    <th
                      aria-label="Name usage: Activate to sort"
                      class="dt-orderable-asc dt-orderable-desc"
                      colspan="1"
                      data-dt-column="1"
                      rowspan="1"
                      tabindex="0"
                    >
                      <span
                        class="dt-column-title"
                        role="button"
                      >
                        Name usage
                      </span>
                      <span
                        class="dt-column-order"
                      />
                    </th>
                    <th
                      aria-label="Family name: Activate to sort"
                      class="dt-orderable-asc dt-orderable-desc"
                      colspan="1"
                      data-dt-column="2"
                      rowspan="1"
                      tabindex="0"
                    >
                      <span
                        class="dt-column-title"
                        role="button"
                      >
                        Family name
                      </span>
                      <span
                        class="dt-column-order"
                      />
                    </th>
                    <th
                      aria-label="Given name: Activate to sort"
                      class="dt-orderable-asc dt-orderable-desc"
                      colspan="1"
                      data-dt-column="3"
                      rowspan="1"
                      tabindex="0"
                    >
                      <span
                        class="dt-column-title"
                        role="button"
                      >
                        Given name
                      </span>
                      <span
                        class="dt-column-order"
                      />
                    </th>
                    <th
                      aria-label="Middle name: Activate to sort"
                      class="dt-orderable-asc dt-orderable-desc dt-type-numeric"
                      colspan="1"
                      data-dt-column="4"
                      rowspan="1"
                      tabindex="0"
                    >
                      <span
                        class="dt-column-title"
                        role="button"
                      >
                        Middle name
                      </span>
                      <span
                        class="dt-column-order"
                      />
                    </th>
                    <th
                      aria-label="Sex: Activate to sort"
                      class="dt-orderable-asc dt-orderable-desc"
                      colspan="1"
                      data-dt-column="8"
                      rowspan="1"
                      tabindex="0"
                    >
                      <span
                        class="dt-column-title"
                        role="button"
                      >
                        Sex
                      </span>
                      <span
                        class="dt-column-order"
                      />
                    </th>
                    <th
                      aria-label="Birth date: Activate to sort"
                      class="dt-orderable-asc dt-orderable-desc dt-right"
                      colspan="1"
                      data-dt-column="9"
                      rowspan="1"
                      tabindex="0"
                    >
                      <span
                        class="dt-column-title"
                        role="button"
                      >
                        Birth date
                      </span>
                      <span
                        class="dt-column-order"
                      />
                    </th>
                    <th
                      aria-label="Dead: Activate to sort"
                      class="dt-orderable-asc dt-orderable-desc dt-type-numeric"
                      colspan="1"
                      data-dt-column="12"
                      rowspan="1"
                      tabindex="0"
                    >
                      <span
                        class="dt-column-title"
                        role="button"
                      >
                        Dead
                      </span>
                      <span
                        class="dt-column-order"
                      />
                    </th>
                    <th
                      aria-label=""
                      class="dt-orderable-none dt-type-numeric"
                      colspan="1"
                      data-dt-column="13"
                      rowspan="1"
                    >
                      <span
                        class="dt-column-title"
                      />
                      <span
                        class="dt-column-order"
                      />
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td
                      class="dt-type-numeric sorting_1"
                    >
                      1
                    </td>
                    <td>
                      string
                    </td>
                    <td>
                      John
                    </td>
                    <td>
                      Doe
                    </td>
                    <td
                      class="dt-type-numeric"
                    />
                    <td>
                      M
                    </td>
                    <td
                      class="dt-right"
                    >
                      21.05.2024
                    </td>
                    <td
                      class="dt-type-numeric"
                    >
                      true
                    </td>
                    <td
                      class="dt-type-numeric"
                    >
                      <div>
                        <a
                          href="/persons/1"
                        >
                          <i
                            class="fa-solid fa-right-to-bracket"
                          />
                        </a>
                      </div>
                    </td>
                  </tr>
                </tbody>
                <tfoot />
              </table>
            </div>
          </div>
          <div
            class="row mt-2 justify-content-between"
          >
            <div
              class="col-md-auto me-auto "
            >
              <div
                aria-live="polite"
                class="dt-info"
                id="DataTables_Table_0_info"
                role="status"
              >
                Showing 1 to 1 of 1 entry
              </div>
            </div>
            <div
              class="col-md-auto ms-auto "
            >
              <div
                class="dt-paging paging_full_numbers"
              >
                <ul
                  class="pagination"
                >
                  <li
                    class="dt-paging-button page-item disabled"
                  >
                    <a
                      aria-controls="DataTables_Table_0"
                      aria-disabled="true"
                      aria-label="First"
                      class="page-link first"
                      data-dt-idx="first"
                      tabindex="-1"
                    >
                      «
                    </a>
                  </li>
                  <li
                    class="dt-paging-button page-item disabled"
                  >
                    <a
                      aria-controls="DataTables_Table_0"
                      aria-disabled="true"
                      aria-label="Previous"
                      class="page-link previous"
                      data-dt-idx="previous"
                      tabindex="-1"
                    >
                      ‹
                    </a>
                  </li>
                  <li
                    class="dt-paging-button page-item active"
                  >
                    <a
                      aria-controls="DataTables_Table_0"
                      aria-current="page"
                      class="page-link"
                      data-dt-idx="0"
                      href="#"
                      tabindex="0"
                    >
                      1
                    </a>
                  </li>
                  <li
                    class="dt-paging-button page-item disabled"
                  >
                    <a
                      aria-controls="DataTables_Table_0"
                      aria-disabled="true"
                      aria-label="Next"
                      class="page-link next"
                      data-dt-idx="next"
                      tabindex="-1"
                    >
                      ›
                    </a>
                  </li>
                  <li
                    class="dt-paging-button page-item disabled"
                  >
                    <a
                      aria-controls="DataTables_Table_0"
                      aria-disabled="true"
                      aria-label="Last"
                      class="page-link last"
                      data-dt-idx="last"
                      tabindex="-1"
                    >
                      »
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
